core/domain/stats_jobs_continuous.py

```diff
@@ -522,29 +522,86 @@ def entity_classes_to_map_over(cls):
 
     @staticmethod
     def map(item):
-        if InteractionAnswerSummariesMRJobManager._entity_created_before_job_queued( # pylint: disable=line-too-long
-                item):
+        def _encode(item):
+            """Encodes a Python value as UTF-8."""
+            if isinstance(item, basestring):
+                return item.encode('utf-8')
+            elif isinstance(item, int) or isinstance(item, float):
+                return item
+            elif isinstance(item, list):
+                return [_encode(subitem) for subitem in item]
+            elif isinstance(item, dict):
+                return {
+                    key: _encode(value) for key, value in item.iteritems()
+                }
+            else:
+                raise Exception('Cannot encode item: %s' % item)
 
-            # All visualizations desired for the interaction.
-            visualizations = (
-                interaction_registry.Registry.get_interaction_by_id(
-                    item.interaction_id).answer_visualizations)
+        # pylint: disable=line-too-long
+        if InteractionAnswerSummariesMRJobManager._entity_created_before_job_queued(
+                item):
+            state_answers_dict = {
+                'exploration_id': item.exploration_id,
+                'exploration_version': item.exploration_version,
+                'state_name': _encode(item.state_name),
+                'interaction_id': item.interaction_id,
+                'submitted_answer_list': item.submitted_answer_list,
+            }
 
-            # Get all desired calculations for the current state interaction id.
-            calc_ids = list(set(viz.calculation_id for viz in visualizations))
-            calculations = [
-                calculation_registry.Registry.get_calculation_by_id(calc_id)
-                for calc_id in calc_ids]
+            # Output answers submitted to the exploration for this exp version.
+            versioned_key = u'%s:%s:%s' % (
+                item.exploration_id, item.exploration_version, item.state_name)
+            yield (versioned_key.encode('utf-8'), state_answers_dict)
 
-            # Perform each calculation, and store the output.
-            for calc in calculations:
-                calc_output = calc.calculate_from_state_answers_entity(item)
-                calc_output.save()
+            # Output the same set of answers independent of the version. This
+            # allows the reduce step to aggregate answers across all
+            # exploration versions.
+            state_answers_dict['exploration_version'] = VERSION_ALL
+            all_versions_key = u'%s:%s:%s' % (
+                item.exploration_id, VERSION_ALL, item.state_name)
+            yield (all_versions_key.encode('utf-8'), state_answers_dict)
 
     @staticmethod
-    def reduce(id, state_answers_model):
-        # pylint: disable=redefined-builtin
-        pass
+    def reduce(key, stringified_submitted_answer_list):
```

Comment:
Sure, done. That somehow was short enough to get rid of the pylint warning for an invalid arg name, too. :)